plutarch-api/src/Plutarch/Api.hs

```diff
@@ -0,0 +1,1275 @@
+{-# LANGUAGE PolyKinds #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE RoleAnnotations #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+{- | = Note
+
+The 'Value.PValue' and 'AssocMap.PMap'-related functionality can be found in
+other modules, as these clash with the Plutarch prelude. These should be
+imported qualified.
+-}
+module Plutarch.Api (
+  -- * Contexts
+  PScriptContext (..),
+  PTxInfo (..),
+  PScriptPurpose (..),
+
+  -- * Tx
+  PTxOutRef (..),
+  PTxOut (..),
+  PTxId (..),
+  PTxInInfo (..),
+  POutputDatum (..),
+
+  -- * Script
+
+  -- ** Types
+  PDatum (..),
+  PDatumHash (..),
+  PRedeemer (..),
+  PRedeemerHash (..),
+  PScriptHash (..),
+
+  -- ** Functions
```

Comment:
There's two "Functions" sections. This needs to be cleaned later once everything lands here

Reply:
They're meant to be per-section.